packages/flame/lib/src/collisions/hitboxes/shape_hitbox.dart

```diff
@@ -4,14 +4,31 @@ import 'package:flame/game.dart';
 import 'package:flame/geometry.dart';
 import 'package:flame/src/geometry/shape_intersections.dart'
     as intersection_system;
-import 'package:meta/meta.dart';
+import 'package:flutter/foundation.dart';
 
 /// A [ShapeHitbox] turns a [ShapeComponent] into a [Hitbox].
 /// It is currently used by [CircleHitbox], [RectangleHitbox] and
 /// [PolygonHitbox].
 mixin ShapeHitbox on ShapeComponent implements Hitbox<ShapeHitbox> {
+  final _collisionType = NotifyingCollisionType(CollisionType.active);
```

Comment:
I would make this public, but `@internal`, so that you don't need the add/remove methods below.

Reply:
But it will become breaking change, because  NotifyingCollisionType is not a subtype of CollisionType enum. This case not similar to NotifyingVector2 for example.

Reply:
Anyway this will lead user to use `collisionTypeNotifier` instead of `collisionType` when changing from "sweep" to "quadtree". It will become addigiotal requirement, different from previous collision system. It seems to me like it is bigger evil for framework design than having two additional functions. Or did I missunderstand you?

Reply:
I think you did misunderstand me, or possibly I misunderstood you. These callbacks should only be sent within the library and not used by the user, right? And if so, the notifier can be made public and marked internal (hidden from the user, but open for us to use internally).

Reply:
Ah, got it! Will fix 👌